duckscript_sdk/src/sdk/std/zip/pack/mod.rs

```diff
@@ -0,0 +1,232 @@
+        };
+
+        let zipfile = Path::new(zipfile);
+        let zipfile_dir = match zipfile.parent() {
+            Some(path) => path,
+            None => return CommandResult::Error("Couldn't parse ZIP file directory.".to_string()),
+        };
+        match std::fs::create_dir_all(zipfile_dir) {
+            Ok(_) => (),
+            Err(err) =>
+                return CommandResult::Error(format!("Couldn't create ZIP file directory: {}", err)),
+        };
+        let zip_file_existed = zipfile.exists();
+        let zip_file = match OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(!zip_file_existed)
+            .truncate(!append)
+            .open(zipfile)
+        {
+            Ok(file) => file,
+            Err(err) =>
+                return CommandResult::Error(format!("Couldn't create/open ZIP file: {}", err)),
+        };
+        let mut zip = if append && zip_file_existed {
+            match ZipWriter::new_append(zip_file) {
+                Ok(writer) => writer,
+                Err(err) =>
+                    return CommandResult::Error(format!("Couldn't open ZIP file: {}", err)),
+            }
+        } else {
+            ZipWriter::new(zip_file)
+        };
+
+        let zip_options = FileOptions::default()
+            .compression_method(compression)
+            .unix_permissions(0o755);
+
+        let mut files = Vec::new();
+        for arg in file_args {
+            match state::get_handle(state, arg.to_string()) {
+                Some(value) =>
+                    match value {
+                        StateValue::List(entries) =>
+                            for entry in entries {
+                                match state::get_as_string(entry) {
+                                    Ok(file) => files.push(file),
+                                    Err(err) => return CommandResult::Error(err),
+                                }
+                            }
+
+                        arg =>
+                            match state::get_as_string(arg) {
+                                Ok(arg) => files.push(arg),
+                                Err(err) => return CommandResult::Error(err),
+                            }
+                    }
+                None => files.push(arg.to_string()),
+            };
+        }
+
+        for file_to_add_str in files {
+            let file_to_add_path = Path::new(&file_to_add_str);
+            let mut file_to_add = match File::open(file_to_add_path) {
+                Ok(file) => file,
+                Err(err) => return CommandResult::Error(
+                    format!("File does not exist or can't be opened: {} ({})",
+                            file_to_add_str,
+                            err)),
+            };
+            let mut file_to_add_contents = Vec::new();
+            match file_to_add.read_to_end(&mut file_to_add_contents) {
```

Comment:
this is an issue. this would break for large files. we need some streaming support i believe, instead of loading everything into the memory.

Reply:
There is actually `std::io::copy` so this was an easy fix